obspy/signal/spectral_estimation.py

```diff
@@ -1958,7 +1958,7 @@ def plot(self, filename=None, show_coverage=True, show_histogram=True,
                 xdata = periods
                 if xaxis_frequency:
                     xdata = frequencies
-                ax.plot(xdata, ydata, '0.4', linewidth=2)
+                ax.plot(xdata, ydata, '0.4', linewidth=2, label=key)
```

Comment:
I don't really like this. You're adding something real to the plot that just stays hidden because we (at the moment) never call `.legend()` on the plot, and that's only because it's used during testing. I don't really understand why you need this, but there has to be another way to do it..

Reply:
OK I also admit, it is kind of cheating...